extensions/mailwip/src/batch-delete-domains.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+  Action,
+  ActionPanel,
+  Alert,
+  Form,
+  Icon,
+  Toast,
+  confirmAlert,
+  openExtensionPreferences,
+  showToast,
+  useNavigation,
+} from "@raycast/api";
+import { useState } from "react";
+import { DOMAINS } from "./utils/constants";
+import { deleteDomains } from "./utils/api";
+import { FormValidation, getFavicon, useForm } from "@raycast/utils";
+import { DomainDelete } from "./utils/types";
+import ErrorComponent from "./components/ErrorComponent";
+
+export default function Domains() {
+  const { push } = useNavigation();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { handleSubmit, itemProps } = useForm<DomainDelete>({
+    async onSubmit(values) {
+      const { domains } = values;
+      if (
+        await confirmAlert({
+          title: `Delete '${domains.join()}'?`,
+          primaryAction: { title: "Delete", style: Alert.ActionStyle.Destructive },
+        })
+      ) {
+        setIsLoading(true);
+        const response = await deleteDomains({ domains });
+        if (!("errors" in response)) {
+          showToast(Toast.Style.Success, response.message);
+          if (
+            await confirmAlert({
+              title: response.message,
+              message: "Go to Extension Preferences to remove domain?",
+              primaryAction: { title: "Yes" },
+            })
+          )
+            openExtensionPreferences();
+        } else {
+          push(<ErrorComponent error={response.errors} />);
+        }
+        setIsLoading(false);
+      }
+    },
+    validation: {
+      domains: FormValidation.Required,
+    },
+  });
+
+  return (
+    <Form
+      isLoading={isLoading}
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm icon={Icon.Check} onSubmit={handleSubmit} />
+          <Action.OpenInBrowser
+            title="Go To API Reference"
+            icon={Icon.Globe}
+            url="https://mailwip.com/api/?javascript#batch-delete-domains"
+          />
+        </ActionPanel>
+      }
+    >
+      <Form.Description text="Batch Delete Domains (you will have to manually remove these from Extension Preferences)" />
```

Comment:
This is not the best UX. Maybe there would be another way to manage the domains instead of leaving them on the preferences?

![CleanShot 2023-07-04 at 15 49 19@2x](https://github.com/raycast/extensions/assets/611002/28a334ed-a5c3-4657-b145-463de0396128)

Reply:
Hi Milena:

Thank you for the detailed review. I agree that this is not the best UX so I will experiment replacing Preferences with a useCachedState() approach and push an update when done.